From fda37c28510f98b48e67a67674b0fcf56d261576 Mon Sep 17 00:00:00 2001
Message-ID: <fda37c28510f98b48e67a67674b0fcf56d261576.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 17 Jan 2024 00:23:30 +0000
Subject: [PATCH 28/38] gnu: go-github-com-minio-blake2b-simd: Move to (gnu
 packages golang-crypto).

* gnu/packages/golang.scm (go-github-com-minio-blake2b-simd): Move from
here ...
* gnu/packages/golang-crypto.scm: ... to here.

Change-Id: I096cf86c48b0fdcbf7c5d0b454d45de12018c908
---
 gnu/packages/golang-crypto.scm | 31 +++++++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 32 --------------------------------
 2 files changed, 31 insertions(+), 32 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 2b2e9d877f..00a6f51313 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -432,6 +432,37 @@ (define-public go-github-com-marten-seemann-qtls
 the Go standard library's TLS 1.3 implementation.")
     (license license:bsd-3)))
 
+(define-public go-github-com-minio-blake2b-simd
+  (let ((commit "3f5f724cb5b182a5c278d6d3d55b40e7f8c2efb4")
+        (revision "0"))
+    (package
+      (name "go-github-com-minio-blake2b-simd")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/minio/blake2b-simd")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0b6jbnj62c0gmmfd4zdmh8xbg01p80f13yygir9xprqkzk6fikmd"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:import-path "github.com/minio/blake2b-simd"))
+      (home-page "https://github.com/minio/blake2b-simd")
+      (synopsis "Fast hashing in pure Go of BLAKE2b with SIMD instructions")
+      (description "This package was initially based on the pure go BLAKE2b
+implementation of Dmitry Chestnykh and merged with the (cgo dependent) AVX
+optimized BLAKE2 implementation (which in turn is based on the official
+implementation.  It does so by using Go's Assembler for amd64 architectures
+with a golang only fallback for other architectures.
+
+In addition to AVX there is also support for AVX2 as well as SSE.  Best
+performance is obtained with AVX2 which gives roughly a 4X performance
+increase approaching hashing speeds of 1GB/sec on a single core.")
+      (license license:asl2.0))))
+
 (define-public go-github-com-minio-sha256-simd
   (package
     (name "go-github-com-minio-sha256-simd")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 581594a1e0..ef20bc067e 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -5674,38 +5674,6 @@ (define-public go-github-com-mreiferson-go-svc
 @url{https://github.com/judwhite/go-svc/raw/master/svc/svc_windows_test.go,here}.")
       (license license:expat))))
 
-(define-public go-github-com-minio-blake2b-simd
-  (let ((commit "3f5f724cb5b182a5c278d6d3d55b40e7f8c2efb4")
-        (revision "0"))
-    (package
-      (name "go-github-com-minio-blake2b-simd")
-      (version (git-version "0.0.0" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://github.com/minio/blake2b-simd")
-               (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32
-           "0b6jbnj62c0gmmfd4zdmh8xbg01p80f13yygir9xprqkzk6fikmd"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:import-path "github.com/minio/blake2b-simd"))
-      (home-page "https://github.com/minio/blake2b-simd")
-      (synopsis "Fast hashing in pure Go of BLAKE2b with SIMD instructions")
-      (description "This package was initially based on the pure go BLAKE2b
-implementation of Dmitry Chestnykh and merged with the (cgo dependent) AVX
-optimized BLAKE2 implementation (which in turn is based on the official
-implementation.  It does so by using Go's Assembler for amd64 architectures
-with a golang only fallback for other architectures.
-
-In addition to AVX there is also support for AVX2 as well as SSE.  Best
-performance is obtained with AVX2 which gives roughly a 4X performance
-increase approaching hashing speeds of 1GB/sec on a single core.")
-      (license license:asl2.0))))
-
 (define-public go-github-com-spaolacci-murmur3
   (package
     (name "go-github-com-spaolacci-murmur3")
-- 
2.41.0

